transport/http2_server.go

```diff
@@ -893,11 +898,11 @@ func (t *http2Server) Write(s *Stream, data []byte, opts *Options) (err error) {
 		// Reset ping strikes when sending data since this might cause
 		// the peer to send ping.
 		atomic.StoreUint32(&t.resetPingStrikes, 1)
-		if err := t.framer.writeData(forceFlush, s.id, false, p); err != nil {
+		if err := t.framer.writeData(forceFlush && !opts.Delay, s.id, false, p); err != nil {
```

Comment:
@apolcyn I agree that this could lead to deadlock. Perhaps it would be sufficient to flush the buffered data whenever `Write` needs to loop. Also, I'm curious what the `forceFlush` logic is doing given that there is another check a few lines below to flush the data if we were the last writer.